core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/aggregation/AvgAggregator.java

```diff
@@ -46,7 +46,7 @@ public AvgState create() {
   public AvgState iterate(BindingTuple tuple, AvgState state) {
     Expression expression = getArguments().get(0);
     ExprValue value = expression.valueOf(tuple);
-    if (!(value.isNull() || value.isMissing())) {
+    if (!(value.isNull() || value.isMissing()) && conditionValue(tuple)) {
```

Comment:
Any chance to move this template code in each subclass to the parent class `Aggregator`? As what I see, each aggregator impl class only differs on how to accumulate state when this condition is true.